### PR TITLE
sql: Improve ergonomics of setting current schema

### LINKED ISSUE #18946: sql: improve ergonomics of setting current schema / `search_path`
*New proposal.* Expanded from @mjibson in https://github.com/MaterializeInc/materialize/issues/18946#issuecomment-1532155294. 

* Allow `SET schema {= | TO} {ident | string}`. This deviates from PostgreSQL, which allows only `SET SCHEMA {string}`, but is consistent with `SET DATABASE` and `SET CLUSTER`.
* Do *not* allow multiple schemas, as in `SET SCHEMA = s1, s2, public`. Users who want to use multiple schemas need to use the `search_path`.
* Add `SHOW schema`, which functions as `SELECT current_schema()`.
* Teach `SHOW database`, `SHOW cluster`, and `SHOW schema` to report a notice if the shown database, cluster, or schema does not exist.

After much discussion, the above proposal is thought to best thread the needle between "match PostgreSQL" and "match user expectations." The weirdest part of the proposal is the fact that `SHOW schema` must show _one_ schema, even if the search path contains multiple schemas. Fortunately, the "current schema" is already a concept in PostgreSQL: it's the schema in which new objects will be created if a schema is not explicitly specified in the DDL statement.

-----

*Initial proposal; superseded.*

~`SCHEMA` is actually an alias for `search_path`.~
~In PG you omit the TO/= when you're using an alias, and you include it when you specify the configuration parameter directly.~

~So you can run `SET SCHEMA foo` or `SET search_path = foo`.~
~This UX is confusing because users typically run `SET DATABASE = bar` first, then expect `SET SCHEMA = foo` to work too. They don't realize SCHEMA is an alias and we don't tell them to use search_path anywhere.~

~When a user types `SET SCHEMA = foo` we should reject it with a hint to type `SET SCHEMA foo` instead.~

### PR DESCRIPTION
**Pull request by jkosh44**
Previously, we followed the PostgreSQL syntax for how to set the current search_path. That is a user could type
`SET search_path = <list-of-schemas>` or `SET schema <single-schema>`. `TO` also works as a replacement for `=` in the first example. Note the lack of an `=` in the second example. The reason PostgreSQL does this is that when using parameter aliases (such as `SCHEMA`) the user has to omit the `=`/`TO`. This differs from how we treat databases and schemas. For those objects users can type
`SET database = <database-name>` or `SET cluster = <cluster-name>` respectively. Many users were confused of the differences between schemas and databases/clusters.

This commit improves the ergonomics of setting the current schema by allowing the syntax `SET schema {= | TO} {ident | string}` which has the same result as `SET schema {string}`. It also adds `SHOW schema` which acts as an alias for `SELECT current_schema()`.

Additionally, `SHOW database`, `SHOW cluster`, and `SHOW schema` emit a notice if they are unable to display a resolvable object.

Resolves #18946

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: test/pgtest-mz/notice.pt
Comment (jkosh44) [inline test/pgtest-mz/notice.pt:1]: This looks wrong

**Review by mjibson — APPROVED** [review]

File: src/adapter/src/notice.rs [hunk 1]
```diff
@@ -106,6 +109,7 @@ impl AdapterNotice {
         match self {
             AdapterNotice::DatabaseDoesNotExist { name: _ } => Some("Create the database with CREATE DATABASE or pick an extant database with SET DATABASE = name. List available databases with SHOW DATABASES.".into()),
             AdapterNotice::ClusterDoesNotExist { name: _ } => Some("Create the cluster with CREATE CLUSTER or pick an extant cluster with SET CLUSTER = name. List available clusters with SHOW CLUSTERS.".into()),
+            AdapterNotice::NoResolvableSearchPathSchema { search_path: _ } => Some("Create a schema with CREATE SCHEMA or pick an extant schema with SET SCHEMA = name. List available schemas with SHOW SCHEMAS".into()),
```
Comment (ParkMyCar) [inline src/adapter/src/notice.rs:112]: nit: add a trailing period
```suggestion
            AdapterNotice::NoResolvableSearchPathSchema { search_path: _ } => Some("Create a schema with CREATE SCHEMA or pick an extant schema with SET SCHEMA = name. List available schemas with SHOW SCHEMAS.".into()),
```

**Review by ParkMyCar — APPROVED** [review]

**Comment by jkosh44** [comment]
Can someone from @MaterializeInc/docs take a look at the added docs? Here's Postgres docs for reference: https://www.postgresql.org/docs/current/sql-set.html

**Comment by jkosh44** [comment]
I'm actually going to merge this and let docs make any needed changes after the fact.

**Pull request merged**